src/js/modules/core/services/sweeper/sweeper-service.factory.js

```diff
@@ -0,0 +1,207 @@
+(function () {
+    "use strict";
+
+    angular.module('blocktrail.core')
+        .factory('sweeperService', SweeperService);
+
+    function SweeperService($q, bitcoinJS, bip39, BlocktrailBitcoinService, launchService) {
+
+        var bitcoinDataClient = null;
+
+        function estimateApproximateFeeP2PKH(inputCount, outputCount) {
+            // https://bitcoin.stackexchange.com/a/46379
+            return inputCount * 146 + outputCount * 33 + 10;
+        }
+
+        function batchDeriveAddressesAndPrivKeys(parentNode, startIndex, batchSize) {
+            var currentNode = null;
+            var currentAddress = null;
+            var batchArray = {};
+
+            for (var i = startIndex; i <= batchSize + startIndex; i++) {
+                currentNode = parentNode.derive(i);
+                currentAddress = currentNode.getAddress();
+                var currentPrivKey = currentNode.keyPair;
+
+                batchArray[currentAddress] = {
+                    priv_key: currentPrivKey,
+                    derive_idx: i
+                }
+            }
+            return batchArray;
+        }
+
+        function batchDiscoverUTXOs(parentNode, batchLoopCount, batchSize) {
+            var batchDataObject = batchDeriveAddressesAndPrivKeys(parentNode, batchLoopCount * batchSize, batchSize);
+            var addresses = Object.keys(batchDataObject);
+
+            return bitcoinDataClient.batchAddressHasTransactions(addresses).then(function (success) {
+                if (!success) {
+                    return false;
+                } else {
+                    return bitcoinDataClient.getBatchUnspentOutputs(addresses).then(function (result) {
```

Comment:
Maybe move the handling part here down, to be run when promise returned by the present function returns the batch of utxos?